src/fetch/contracts.py

```diff
@@ -66,21 +59,30 @@ def _limited_is_contract(self, addresses: list[str]) -> dict[str, bool]:
         results = {}
         for result_dict in response.json():
             try:
-                results[addresses[result_dict['id']]] = result_dict['result'] != "0x"
+                results[addresses[result_dict['id']]] = result_dict['result']
             except KeyError as err:
                 raise IOError(
-                    f"Request for address \"{addresses[result_dict['id']]}\" "
+                    f"Request for code at address \"{addresses[result_dict['id']]}\" "
                     f"failed with response {result_dict}"
                 ) from err
         return results
 
     def load_from_file(self, file: File) -> dict[str, bool]:
         """Loads results dict from a file containing known contract addresses"""
-        print(f"Loading Contracts from {file.name}")
+        print(f"loading contracts from {file.name}")
         with open(file.filename(), 'r', encoding='utf-8') as txt_file:
             contracts = set(txt_file.read().splitlines())
         return {addr: addr in contracts for addr in self.addresses}
 
+    def batch_call(self, addresses: list[str], func):
```

Comment:
Since `batch_call` uses the result of the `func` call it would be nice to annotate the type accordingly.
I'm guessing `func` takes a `list[str]` and returns a map of `str -> str`?

Reply:
I would need to figure out how to annotate function types...

Reply:
I found [this](https://stackoverflow.com/a/53227734) not sure if that is applicable because it mentions using a module for that.

Reply:
We can import from typing (it's part of the standard library) I'll try it out soon.